projects/bin/src/main/kotlin/bin/SerialPort.kt

```diff
@@ -19,7 +19,10 @@ internal data class SerialPort(val name: String, val baudRate: Int) {
     fun recvChar(): Char = recvByte().toChar()
 
     fun send(bytes: ByteArray) {
-        serialPort.writeBytes(bytes, bytes.size.toLong())
+        val sentCount = serialPort.writeBytes(bytes, bytes.size.toLong())
+        if (sentCount != bytes.size) {
```

Comment:
This should be split into two assertions, one for `-1` and another for a mismatch.